src/http-adapter.js

```diff
@@ -109,39 +110,36 @@ class HttpAdapter {
     }
     promise.then(removeFromOutstandingRequests, removeFromOutstandingRequests)
 
-    return promise
+    return await promise
   }
 
-  _makeRequest ({ url, type, headers, data }) {
-    return new Promise((resolve, reject) => {
-      let request = {
-        url,
-        type,
-        headers,
-        success: (data, textStatus, jqXhr) => {
-          if (!data && jqXhr.status !== 204) {
-            throw new Error(`request returned ${jqXhr.status} status without data`)
-          }
-          return resolve(data)
-        },
-        error: (response) => {
-          if (response.readyState === 0 || response.status === 0) {
-            // this is a canceled request, so we literally should do nothing
-            return
-          }
-
-          const error = new Error(`request for resource, ${url}, returned ${response.status} ${response.statusText}`)
-          error.response = response
-          reject(error)
-        },
-        // being explicit about data type so jQuery doesn't "intelligent guess" wrong
-        // changing this may not break tests, but does behave badly in prod
-        dataType: 'text',
-        data
-      }
-
-      ajax(request)
+  async _fetch ({method, url, headers, body}) {
+    let response = await fetch(url, {
+      method,
+      url,
+      headers,
+      ...(body || {})
```

Comment:
Maybe you meant:
```suggestion
      body: body || {}
```
From our convo f2f, `body` can be used for either `URLSearchParams` or a JSON body.

Reply:
But if you default `body` to `{}` in the function params, you can skip the default...